cloudbuild.yaml

```diff
@@ -19,10 +20,15 @@ steps:
     args:
       - buildx
       - build
-      - --platform="linux/amd64,linux/arm64"
+      - --platform=linux/amd64,linux/arm64
       - --tag=gcr.io/k8s-staging-sig-storage/objectstorage-controller:${_GIT_TAG}
+      # using _PULL_BASE_REF as a tag will often just build and overwrite the same 'master' tag,
+      # BUT! if the commit has a git tag, it will build that tag instead. this mechanism allows
+      # creating the semver-tagged images that will be auto-promoted to release
+      - --tag=gcr.io/k8s-staging-sig-storage/objectstorage-controller:${_PULL_BASE_REF}
```

Comment:
Here's the easy strategy for building git tagged commits I'm hoping will work.

Reportedly, for _PULL_BASE_REF:
> _PULL_BASE_REF will contain the base ref that was pushed to - for instance, master or release-0.2 for a PR merge, or v0.2 for a tag. You can use this for logic in your build process (like deciding whether to update latest) if desired.

Reply:
From the logs, this is what the custom subs were for the last, failed build:
`_PULL_BASE_REF=master,_GIT_TAG=v20230914-v0.1.1-23-gaef2d3b`

IMO, the `_GIT_TAG` is a perfectly usable staging tag. The combo of date and number of commits past the tag (`-23-`) means each build should always follow the preceding one when sorting alphabetically.

Pushing a `master` or other <branch> tag seems fine to me. `<image>:master` would effectively be a synonym for `<image>:latest`. If COSI starts using `release-X` branches, then `<image>:release-X` would be similar to `latest` but for a particular release branch.

For now, I think it's easy to stick to a continuous release model, releasing from master and `git tag`-ing (GitHub release-ing) new versions.